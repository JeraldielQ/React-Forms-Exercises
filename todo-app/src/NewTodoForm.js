import React, { useState } from "react";

const NewTodoForm = ({ addToDo }) => {
  const INITIAL_STATE = { ToDo: "" };
  const [todoData, setToDoData] = useState(INITIAL_STATE);

  const handleSubmit = evt => {
    evt.preventDefault();
    addToDo(todoData);
    setToDoData(INITIAL_STATE);
  };

  const handleChange = evt => {
    const { name, value } = evt.target;
    setToDoData(tData => ({
      ...tData,
      [name]: value
    }));
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="ToDo">ToDo:</label>
      <input
        id="ToDo"
        name="ToDo" 
        value={todoData.ToDo} 
        onChange={handleChange}
      />

      <button>Add a new item!</button>
    </form>
  );
};

export default NewTodoForm;
