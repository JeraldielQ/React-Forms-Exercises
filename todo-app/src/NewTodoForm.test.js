import React from "react";
import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders without crashing", function() {
  render(<NewTodoForm addToDo={() => {}} />);
}); 

it("matches snapshot", function() {
    const { asFragment } = render(<NewTodoForm addToDo={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });