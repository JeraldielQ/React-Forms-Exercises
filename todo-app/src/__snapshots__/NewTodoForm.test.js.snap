// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="ToDo"
    >
      ToDo:
    </label>
    <input
      id="ToDo"
      name="ToDo"
      value=""
    />
    <button>
      Add a new item!
    </button>
  </form>
</DocumentFragment>
`;
